### DISCUSSION TITLE
UCLA Miniscope Triggering via Bonsai

### DISCUSSION THREAD

**Discussion by AshhadQazi** [bonsai-rx/bonsai#2056] [Q&A]
Hello!

I need some assistance with triggering the UCLA Miniscope through Bonsai. Currently, we've set up a system using HARP to trigger the accelerometer, a FLIR Blackfly camera, and the UCLA Miniscope. The main triggering hub is the HARP BaseStation, which sends out a signal to the other devices once the accelerometer starts collecting data. This setup has been working well so far, but we're running into an issue when trying to integrate the UCLA Miniscope into the triggering workflow.

![image](https://github.com/user-attachments/assets/8787ab43-6787-46bf-83fa-e4fd1a71c322)

Here’s the issue: when we press "Start" in Bonsai, the Miniscope starts capturing video data before we press "Q" to begin acquiring accelerometer data. This leads to a problem where the Miniscope starts recording prematurely. To address this, we’ve set the "Triggerable" option in the DAQ software as shown below, and we’re able to trigger and save Miniscope video data. However, when doing this, the video node in Bonsai remains blank, and no input is being received from the UCLA Miniscope node.

![image](https://github.com/user-attachments/assets/6d46818c-94d3-4776-9296-9814b7e991a0)

Although this setup works for capturing video, it bypasses Bonsai entirely, meaning we lose the ability to process the video data within the Bonsai framework, which we were hoping to use for further analysis.

What we’re looking for is a way to trigger the UCLA Miniscope through Bonsai, as we’ve done with the other devices, so that we can capture and process the video data within the Bonsai environment.

If anyone has experience with this or suggestions on how to properly integrate the UCLA Miniscope into the Bonsai triggering workflow, I would greatly appreciate your help!

Thank you!

[UCLA_Miniscope111324.zip](https://github.com/user-attachments/files/17782386/UCLA_Miniscope111324.zip)

**Comment by bruno-f-cruz** [comment]
Try SubscribeWhen instead of SkipUntil. That should delay the subscription and prevent the device from acquiring. You must place the subscription node before the branching point.

Hope it helps
